Zilon.Core/Zilon.Core/PersonModules/EquipmentModule.cs

```diff
@@ -20,24 +24,34 @@ public EquipmentModule(IReadOnlyCollection<PersonSlotSubScheme> slots) : base(sl
 
         protected override void ValidateSetEquipment(Equipment equipment, int slotIndex)
         {
-            var slot = Slots[slotIndex];
+            var slot = GetSlotForEquipByIndex(equipment: equipment, slotIndex: slotIndex);
 
-            if (!EquipmentCarrierHelper.CheckSlotCompability(equipment, slot))
+            if (!EquipmentCarrierHelper.CheckSlotCompability(equipment: equipment, slot: slot))
             {
                 throw new ArgumentException(
                     $"Для экипировки указан слот {slot}, не подходящий для данного типа предмета {equipment}.");
             }
 
-            if (!EquipmentCarrierHelper.CheckDualCompability(this, equipment, slotIndex))
+            if (!EquipmentCarrierHelper.CheckDualCompability(equipmentModule: this, equipment: equipment,
+                slotIndex: slotIndex))
             {
                 throw new InvalidOperationException(
                     $"Попытка экипировать предмет {equipment}, несовместимый с текущий экипировкой.");
             }
 
-            if (!EquipmentCarrierHelper.CheckShieldCompability(this, equipment, slotIndex))
+            if (!EquipmentCarrierHelper.CheckShieldCompability(
+                equipmentCarrier: this,
+                equipment: equipment,
+                slotIndex: slotIndex))
             {
                 throw new InvalidOperationException("Попытка экипировать два щита.");
             }
         }
+
+        private PersonSlotSubScheme GetSlotForEquipByIndex(Equipment equipment, int slotIndex)
+        {
+            return Slots.ElementAtOrDefault(slotIndex) ?? throw new ArgumentException(
+                $"Попытка экипировать предмет {equipment} в несуществующий слот {slotIndex}");
```

Comment:
Please, try to translate everything you can in english:
"Attempting to equip an item {equipment} in a nonexistent {slotIndex} slot."

It is good to open-source projects.